docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+
+If the supplied patient age is a word or is missing, the age will be set to ```-1```. This value is chosen to indicate 
+that there isn’t a valid age set. Thus, when displaying the age, if ```-1``` is encountered, show age as an empty string 
+instead.
+
+Below shows the sequence diagram for ```AddPatientCommand``` class
+
+![](images/AddPatientCommandSequence.png)
+ 
+
+##### 2.2.4.2 AddAppointmentClass
+
+To add an appointment, the ```AddAppointmentCommand``` class is used. For this ```AddAppointmentCommand``` class, it 
+serves as a façade class for the ```Main```, ```Appointment```, ```AppointmentList``` and the ```Storage``` class to 
+interact with one another. 
+
+![](images/AddAppointmentDiagram.png)
+
+1. Like the ```AddPatientCommand``` class, the ```AddAppointmentCommand``` object is first created by the ```Parser``` 
+object, where the information of the appointment is again stored in a Map that the ```AddAppoinmentCommand``` object would read from. 
+Said information will be stored in the ```AddAppoinmentCommand``` object
+
+2. When 
+the ```Main``` calls ```execute(Ui ui, Storage storage)```, the ```AddAppointmentCommand``` class would call upon the 
+```Appointment``` class to make an ```Appointment``` Object. 
+
+3. After which, the ```AddAppoinmentCommand``` object will 
+call upon the ```AppointmentList``` object to obtain the list of ```Appointments``` (get the ```List``` object that 
+represents the list of appointments by ```AppointmentList```’s ```getAppointmentList()``` command) so that it can 
+directly add the new ```Appointment``` object into the appointment list. 
+
+4. Finally, it will call upon the ```Storage``` 
+class’s ```saveAppoinmentList()``` function to save the updated appointment list. 
+
+5. Upon successfully adding the 
+```Appointment``` object into the appointment list, it will call upon the ```Ui``` class’ 
+```showAppointmentAddSuccess()``` function to display the success of adding the ```appointment``` into the appointment 
+list.
+
+Below shows the sequence diagram for ```AddAppointmentCommand``` class
+
+![](images/AddAppointCommandSequence.png)
+
+##### 2.2.4.3 ListPatientClass
+
+To display the list of patients, the ```ListPatientCommand``` class is called. This class serves as a façade class of 
+```Main``` and ```Ui``` to interact with each other. 
+
+![](images/ListPatientDiagram.png)
+
+1. This class' object is first created by the ```Parser``` class, where it is then returned to the ```Main``` class to have its
+```execute(Ui ui, Storage storage)``` function be called. 
+
+2. When the ```Main``` class calls the 
+```execute(Ui ui, Storage storage)``` function, ```ListPatientCommand``` will call upon the ```Ui```’s 
+```showEntirePatientList()``` function to display the list of patients.
+
+Below shows the sequence diagram for ```ListPatientCommand``` class
+
+![](images/ListPatientCommandSequence.png)
```

Comment:
![image](https://user-images.githubusercontent.com/31772358/77993888-9fe49f80-735b-11ea-837e-bc3bd4fec704.png)

Same here: should there be a cut at the lower right corner of the loop rectangle?